instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -41,54 +40,29 @@
   @Override
   public void beforeExecution(
       final Context.BeforeExecution context, final ExecutionAttributes executionAttributes) {
-    Span span =
-        ClientDecorator.getOrCreateSpan(DECORATE.spanName(executionAttributes), AwsSdk.tracer());
-    DECORATE.afterStart(span);
+    Span span = TRACER.getOrCreateSpan(TRACER.spanName(executionAttributes), AwsSdk.tracer());
```

Comment:
Do you really need to call `spanName` separately? Why `getOrCreateSpan` cannot handle this?

Reply:
We don't have the request.

Reply:
What do you mean? You can pass `executionAttributes` into `TRACER.getOrCreateSpan`, no? Tracer's `startSpan` method (or its analogue) can accept all data that is needed for him to create a new span. In this case tracer needs `ExecutionAttributes` to deduce span name.

Reply:
getOrCreateSpan takes Tracer as one of the input parameters.
`public Span getOrCreateSpan(String name, Tracer tracer) `

Can't use startSpan because getting the span is needed in beforeExecution and onRequest is called inside afterMarshalling.  ExecutionInterceptor has its own lifecycle.

Reply:
@iNikem are you suggesting adding `startSpan` overload in `AwsSdkClientTracer` that also accepts `executionAttributes`?

Reply:
This `getOrCreateSpan` is already defined in `AwsSdkClientTracer` and used only from here. Just pass `executionAttributes` as first parameter and that's it, API surface is already smaller.

Also, `AwsSdk.tracer()` should not go here, it should go into the constructor of the tracer. There is `io.opentelemetry.instrumentation.api.decorator.BaseTracer#tracer` for that.

Reply:
move spanName helper method to interceptor class.

Reply:
I beg your pardon? My intention was to move as much code as possible from this interceptor into the tracer. You did the exact opposite??

Reply:
I think this is related to my comment here

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/918#discussion_r468512189

I found it hard to understand the tracer when it had so much of the AWS SDK lifecycle copied into it.

Reply:
this spanName method is customizing the span name.  It is only used in the interceptor and nowhere else.  
I also moved a few other helper methods related to AWS SDK lifecycle to the interceptor class.  
@trask has created an issue to clean up AwsSdkClientTracer.  I think we all agreed that AWS SDK doesn't fit well with the Tracer pattern.